### PR TITLE
Basic world shadowing

### PR DESCRIPTION
**Pull request by jonas-schievink**
Plants are not yet affected by shadows (but they *do* cast shadows). We'll do that later, after we wrote a fragment shader we like.

### CODE REVIEW

**Comment by jonas-schievink** [comment]
Oh oops, I accidentally the whole branch :bomb: :boom:

**Comment by jonas-schievink** [comment]
Yay for `reflog`!

File: client/shader/chunk_std.frag [hunk 1]
```diff
@@ -1,12 +1,47 @@
...
+            shadowFactor += shadow ? 1.0 : 0.0;
+        }
+    }
+
+    // Height/Width of the square we sample for Percentage Closer Filtering
+    // (in Pixels)
+    const int PCF_PIXELS = 1 + 2 * SHADOW_PCF_RADIUS;
+
+    // Divide by number of pixels we sampled, to get  a range from 0 to 1
+    shadowFactor /= PCF_PIXELS * PCF_PIXELS;
```
Comment (LukasKalbertodt) [inline client/shader/chunk_std.frag]: `PCF_PIXELS` is a strange name when it holds the square root of the number of all PCF pixels... but yeah ok

**Comment by LukasKalbertodt** [comment]
You're using a real depth framebuffer, which OpenGL writes`gl_Position.z/gl_Position.w` to. This means the depth distribution is logarithmic, which could decrease accuracy at the back plane. However, this might not matter at all...

File: client/shader/chunk_std.frag [hunk 2]
```diff
@@ -1,12 +1,47 @@
...
 in vec3 x_color;
 in vec3 surfaceNormal;
 
 out vec3 color;
 
+// FIXME This should be a `sampler2DShadow`, but glium doesn't expose it
+uniform sampler2D shadow_map;
+// Shadow map height/width in pixels:
+// TODO Set from Rust code so it's always consistent
+uniform float shadow_map_size = 1024;
```
Comment (LukasKalbertodt) [inline client/shader/chunk_std.frag]: I think `textureSize(shadow_map, 0)` should return the correct resolution.

**Pull request merged**